File: MinSum/minsum.js
/*
 * Problem statement: Given a matrix of m rows and n columns. You need to find minimum of each rows and return the sum of those elements
 */

const findMinFromEachRow = (matrix) => {
  const minElements = [];
  for (let i = 0; i < matrix.length; i++) {
    let minValue = Number.MAX_VALUE;
    for (let j = 0; j < matrix[i].length; j++)
      if (matrix[i][j] < minValue) minValue = matrix[i][j];
    minElements.push(minValue);
  }

  let sumValue = 0;
  for (const el of minElements) sumValue += el;
  return sumValue;
};

const matrix = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
];
const minsum = findMinFromEachRow(matrix);
console.log(minsum); // Output: 12

File: MinSum/Minsum.java
package MinSum;

/*
 * Problem statement: Given a matrix of m rows and n columns. You need to find
 * minimum of each rows and return the sum of those elements
 */

import java.util.*;

public class Minsum {

  public static void main(String[] args) {
    var sc = new Scanner(System.in);
    int rows = sc.nextInt();
    int cols = sc.nextInt();
    int[][] matrix = new int[rows][cols];
    for (int i = 0; i < rows; i++) {
      for (int j = 0; j < cols; j++)
        matrix[i][j] = sc.nextInt();

    }

    int[] minElements = findMinEachRow(matrix);
    int minsum = 0;
    for (int el : minElements)
      minsum += el;
    System.out.println(minsum);

    sc.close();
  }

  private static int[] findMinEachRow(int[][] matrix) {
    int[] minElements = new int[matrix.length];
    for (int i = 0; i < matrix.length; i++) {
      int minValue = Integer.MAX_VALUE;
      for (int j = 0; j < matrix[i].length; j++)
        if (matrix[i][j] < minValue)
          minValue = matrix[i][j];
      minElements[i] = minValue;
    }
    return minElements;
  }
}